Tasks/Npm/npmtask.ts

```diff
@@ -4,19 +4,38 @@ import path = require('path');
 import url = require('url');
 import tl = require('vsts-task-lib/task');
 import trm = require('vsts-task-lib/toolrunner');
+var nutil = require('nuget-task-common/utility.js');
+
 var extend = require('util')._extend;
 
 interface EnvironmentDictionary { [key: string]: string; }
 
-tl.setResourcePath(path.join( __dirname, 'task.json'));
+tl.setResourcePath(path.join(__dirname, 'task.json'));
 
 executeTask();
 
 async function executeTask() {
 
-    var cwd = tl.getPathInput('cwd', true, false);
-    tl.mkdirP(cwd);
-    tl.cd(cwd);
+    try {
+        var filePath = tl.getPathInput('cwd', true, false);
+        var dirList = [];
+        var filesList = nutil.resolveFilterSpec(filePath, tl.getVariable("System.DefaultWorkingDirectory"));
+        tl.debug("number of files matching filePath: " + filesList.length);
+        for (var workingDir of filesList) {
+            if (tl.stats(workingDir).isFile()) {
+                workingDir = path.dirname(workingDir);
+            }
+            if (dirList.indexOf(workingDir) === -1) {
+                dirList.push(workingDir);
+            }
+        }
+    }
+    catch (error) {
+        tl.warning(error);
+    }
+    if (dirList.length === 0) {
```

Comment:
Instead of this, should you do this only when the input is empty directory. Otherwise if the pattern match doesn't yield anything, then we should not run it in the default directory.